app/helpers/discovered_hosts_helper.rb

```diff
@@ -75,4 +75,16 @@ def discovery_status_icon(host)
     "<span class='glyphicon #{status_glyph}' style='color: #{status_color}'
       title='#{status_message}'/>".html_safe
   end
+
+  def host_taxonomy_select(f, taxonomy)
+    # Add hidden field with taxonomy value to be updated by the form on submit
+    tax_html = super
+    tax_id = "#{taxonomy.to_s.downcase}_id"
+    selected_taxonomy = params[:host][tax_id]
+    if @override_taxonomy && selected_taxonomy
+      hidden_tax = f.hidden_field tax_id.to_sym, :value => selected_taxonomy
+      tax_html += hidden_tax
+    end
+    tax_html
+  end
```

Comment:
It's a hack into host edit form, I add a hidden input field to it that will send org and loc properties to host#update process, so it will update them too.